cpp/src/cylon/util/arrow_utils.cpp

```diff
@@ -376,5 +376,16 @@ arrow::Status MakeEmptyArrowTable(const std::shared_ptr<arrow::Schema> &schema,
   return arrow::Status::OK();
 }
 
+bool CheckArrowTableContainsChunks(const std::shared_ptr<arrow::Table> &table,
+                                   const std::vector<int> &columns) {
+  if (columns.empty()) {
+    return std::any_of(table->columns().begin(), table->columns().end(),
+                       [](const auto &col) { return col->num_chunks() > 1; });
+  } else {
+    return std::any_of(columns.begin(), columns.end(),
+                       [&](const auto &i) { return table->column(i)->num_chunks() > 1; });
```

Comment:
I think it would be better to use `int` (or `auto`) here
```suggestion
                       [&](int i) { return table->column(i)->num_chunks() > 1; });
```